### PR TITLE
feat: new onramp flow progress

### PR DESCRIPTION
**Pull request by ChefBingbong**
---

## PR-Codex overview
This PR removes unused files and components related to on-ramp providers, updates text components, refactors styles, adds a new button component, and adjusts menu configurations.

### Detailed summary
- Removed unused files and components related to on-ramp providers
- Updated text components and added a new button component
- Refactored styles in certain components
- Adjusted menu configurations and added new icons for shopping basket functionality

> The following files were skipped due to too many changes: `apps/web/src/views/BuyCrypto/utils/NativeBtc.ts`, `apps/web/src/views/BuyCrypto/components/FAQ.tsx`, `apps/web/src/state/buyCrypto/actions.ts`, `apps/web/src/components/Menu/UserMenu/WalletInfo.tsx`, `apps/web/src/components/FiatOnRampModal/ProviderIframe.tsx`, `packages/uikit/src/components/SubMenuItems/SubMenuItems.tsx`, `apps/web/src/components/SearchModal/CurrencySearchModal.tsx`, `apps/web/src/__tests__/utils/utils.test.ts`, `packages/uikit/src/components/Svg/Icons/ShoppingBasket.tsx`, `packages/uikit/src/components/Svg/Icons/ShoppingBasketFilled.tsx`, `apps/web/src/views/BuyCrypto/hooks/useBitcoinAddressValidator.ts`, `apps/web/src/views/BuyCrypto/hooks/useOnRampLimits.ts`, `apps/web/src/views/BuyCrypto/components/OnRampProviderLogo/OnRampProviderLogo.tsx`, `apps/web/src/views/BuyCrypto/hooks/useOnRampSignature.ts`, `apps/web/src/views/BuyCrypto/hooks/useOnRampQuotes.ts`, `apps/web/src/views/BuyCrypto/hooks/useOnRampInputError.ts`, `apps/web/src/views/BuyCrypto/types.ts`, `apps/web/src/state/buyCrypto/reducer.ts`, `packages/uikit/src/__tests__/widgets/__snapshots__/menu.test.tsx.snap`, `apps/web/src/views/BuyCrypto/components/ProviderSelector/ProviderGroupItem.tsx`, `apps/web/src/views/BuyCrypto/components/ProviderSelector/ProviderSelector.tsx`, `apps/web/src/components/SearchModal/CurrencySearch.tsx`, `apps/web/src/components/SearchModal/CurrencyList.tsx`, `apps/web/src/components/SearchModal/OnRampCurrencySearch.tsx`, `apps/web/src/components/SearchModal/OnRampCurrencyList.tsx`, `apps/web/src/views/BuyCrypto/components/TransactionFeeDetails/TransactionFeeDetails.tsx`, `apps/web/src/components/SearchModal/OnRampCurrencyModal.tsx`, `apps/web/src/views/BuyCrypto/styles.tsx`, `apps/web/src/views/BuyCrypto/components/OnRampCurrencySelect/index.tsx`, `apps/web/src/state/buyCrypto/hooks.ts`, `apps/web/src/views/BuyCrypto/constants.ts`, `apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx`, `apps/web/src/views/BuyCrypto/containers/BuyCryptoForm.tsx`, `packages/localization/src/config/translations.json`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/components/Menu/config/config.ts [hunk 1]
```diff
@@ -88,18 +89,30 @@ const config: (
...
+      items: [
+        {
+          label: t('Buy Crypto'),
+          href: '/buy-crypto',
+        },
+        // {
+        //   label: t('Sell Crypto'),
+        //   href: '/sell-crypto',
+        //   disabled: true,
+        // },
```
Comment (chefjackson) [inline apps/web/src/components/Menu/config/config.ts:113]: please remove if not needed

File: apps/web/src/config/constants/endpoints.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export const SNAPSHOT_API = `${SNAPSHOT_BASE_URL}/graphql`
 export const SNAPSHOT_HUB_API = `${SNAPSHOT_BASE_URL}/api/message`
 export const GRAPH_API_POTTERY = 'https://api.thegraph.com/subgraphs/name/pancakeswap/pottery'
 // export const ONRAMP_API_BASE_URL = 'https://pcs-onramp-api.com'
-export const ONRAMP_API_BASE_URL = 'https://onramp2-api.pancakeswap.com'
+export const ONRAMP_API_BASE_URL = 'https://sea-turtle-app-8wv84.ondigitalocean.app'
```
Comment (chefjackson) [inline apps/web/src/config/constants/endpoints.ts:13]: why are we using testing endpoint?

File: apps/web/src/hooks/useCanBuyCrypto.ts [hunk 3]
```diff
@@ -1,21 +1,15 @@
...
 import { useMemo } from 'react'
 
-// TODO: refactor. We shouldn't import modules from views in higher level hooks
-import { SUPPORTED_ONRAMP_TOKENS, isBuyCryptoSupported } from 'views/BuyCrypto/constants'
+const SUPPORTED_ONRAMP_TOKENS = ['ETH', 'DAI', 'USDT', 'USDC', 'BUSD', 'BNB', 'WBTC']
 
 interface Params {
   currencySymbol?: string
-  chainId?: ChainId
+  chainId?: number
```
Comment (chefjackson) [inline apps/web/src/hooks/useCanBuyCrypto.ts:7]: any reason not using enum?

File: packages/chains/src/chainNames.ts [hunk 4]
```diff
@@ -23,6 +23,29 @@ export const chainNames: Record<ChainId, string> = {
...
+  [ChainId.ARBITRUM_SEPOLIA]: 'Arbitrum Sepolia',
+  [ChainId.SCROLL_SEPOLIA]: 'Scroll Sepolia',
+  [ChainId.LINEA]: 'Linea',
+  [ChainId.LINEA_TESTNET]: 'Linea Testnet',
+  [ChainId.BASE]: 'Base',
+  [ChainId.BASE_TESTNET]: 'Base Testnet',
+  [ChainId.BASE_SEPOLIA]: 'Base Sepolia',
+  [ChainId.SEPOLIA]: 'Sepolia',
+}
+
```
Comment (chefjackson) [inline packages/chains/src/chainNames.ts:48]: I'm not sure if it's a good idea to maintain the name in the package. How about we reuse the chain names we have for network switcher?

https://github.com/pancakeswap/pancake-frontend/blob/develop/apps/web/src/components/NetworkSwitcher.tsx#L55-L66

File: apps/web/src/config/constants/endpoints.ts (hunk 2 above)
Comment (ChefBingbong) [reply apps/web/src/config/constants/endpoints.ts]: was just testing with it. reverted

File: apps/web/src/hooks/useCanBuyCrypto.ts (hunk 3 above)
Comment (ChefBingbong) [reply apps/web/src/hooks/useCanBuyCrypto.ts]: yep changee

**Review by chefjackson — APPROVED** [review]

**Pull request merged**